### PR TITLE
Chore/config

### PR DESCRIPTION
**Pull request by danieleades**
adds clippy and rustfmt config

the rustfmt config file is lifted straight from another project of mine, so I expect you'll have some opinions about which config to keep, and which to remove. Normalising comments is nice to have. Import coalescing and reordering tend to be more divisive. What do you think?

the clippy config file sets the MSRV. This prevents clippy from making suggestions involving syntax which is not compatible with the MSRV

update: I've also updated the CI script

- install toolchain in a consistent way across jobs
- use nightly toolchain for clippy and rustfmt

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -83,14 +85,13 @@ jobs:
     steps:
       - name: checkout_repository
         uses: actions/checkout@v2.4.0
-        with:
-          fetch-depth: 1
       - name: install_rust
-        uses: hecrj/setup-rust-action@v1
```
Comment (foresterre) [inline .github/workflows/ci.yml:89]: Let's split these CI workflow changes to another PR; I've used both setup actions, and both work fine. Altough, the nice thing of the actions-rs one is that it allows using a minimal profile, (not sure about Hecrj's action), and is probably more actively maintained (but again, not sure here). Taking these things into account, I'm not sure whether it's a change I would want to make, but I have nothing against it either.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -105,7 +106,7 @@ jobs:
       - name: install_rust
         uses: actions-rs/toolchain@v1
         with:
-          toolchain: stable
+          toolchain: nightly
```
Comment (foresterre) [inline .github/workflows/ci.yml:109]: Can I assume the change in clippy toolchain is because of a configuration setting which is not yet on the stable channel? I would prefer to stay on stable, as there have been some suggestions in the past, added to the default category, which would have been more reasonable in the pendantic category. Since builds will fail when new clippy versions are released (this is a good thing in my opinion), which requires extra attention, I would like to keep clippy changes which may not be that important to a minimum.

File: .rustfmt.toml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+version = "Two"
```
Comment (foresterre) [inline .rustfmt.toml:1]: Would it be possible to split the format changes to a separate PR as well (i.e. leaving just the clippy config in this PR)? That would also allow me to play around with the settings a bit more, before deciding on what I would like to keep and what not.

File: .clippy.toml [hunk 4]
```diff
@@ -0,0 +1 @@
+msrv = "1.53.0"
```
Comment (foresterre) [inline .clippy.toml:1]: 👍🏼

**Review by foresterre — CHANGES_REQUESTED** [review]

**Comment by danieleades** [comment]
@foresterre thanks for the feedback. I'll split this up into separate PRs.

i'll close for now
